desc/experimental/_terpsichore.py

```diff
@@ -0,0 +1,872 @@
+    f.write("           0           0           0           0\n")
+    f.write("      LLHSPR      LRHSPR      LEIGPR      LEFCPR\n")
+    f.write("           9           9           1           1\n")
+    f.write("      LXYZPR      LIOTPL      LDW2PL      LEFCPL\n")
+    f.write("           0           1           1           1\n")
+    f.write("      LCURRF      LMESHP      LMESHV      LITERS\n")
+    f.write("           1           1           2           1\n")
+    f.write("      LXYZPL      LEFPLS      LEQVPL      LPRESS\n")
+    f.write("           1           1           0           0\n")
+    f.write("C\n")
+    f.write(
+        "C    PVAC        PARFAC      QONAX        QN         DSVAC       "
+        + "QVAC    NOWALL\n"
+    )
+    f.write(
+        "  1.0001e+00  0.0000e-00  0.6500e-00  0.0000e-00  1.0000e-00  "
+        + "1.0001e+00     -2\n"
+    )
+    f.write("\n")
+    f.write(
+        "C    AWALL       EWALL       DWALL       GWALL       DRWAL       DZWAL   "
+        + "NPWALL\n"
+    )
+    f.write(
+        "  {:10.4e}  ".format(awall)
+        + "1.5000e+00 -1.0000e-00  5.2000e-00 -0.0000e-00 +0.0000e-00      2\n"
+    )
+    f.write("C\n")
+    f.write("C    RPLMIN       XPLO      DELTAJP       WCT        CURFAC\n")
+    f.write(
+        "  1.0000e-05  1.0000e-06  {:10.4e}  6.6667e-01  1.0000e-00\n".format(deltaJp)
+    )
+    f.write("C\n")
+    f.write(
+        "C                                                             "
+        + "MODELK =      {}\n".format(modelk)
+    )
+    f.write("C\n")
+    f.write(
+        "C     NUMBER OF EQUILIBRIUM FIELD PERIODS PER STABILITY PERIOD: "
+        + "NSTA =      {}\n".format(nfp)
+    )
+    f.write("C\n")
+    f.write("C     TABLE OF FOURIER COEFFIENTS FOR STABILITY DISPLACEMENTS\n")
+    f.write("C\n")
+    f.write(
+        "C M=  0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 "
+        + "0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5  N\n"
+    )
+
+    for n in range(N_min, N_max + 1):
+        mode_str = "     "
+        for m in range(56):
+            if mode_family < 0 or n % 2 == mode_family:  # FIXME: modify for nfp != 2,3
+                if (m <= M_max) and (n >= N_min) and (n <= N_max):
+                    if m == 0:
+                        if n > 0:
+                            mode_str += " 1"
+                        else:
+                            mode_str += " 0"
+                    else:
+                        mode_str += " 1"
+                else:
+                    mode_str += " 0"
+            else:
+                mode_str += " 0"
+
+        mode_str += "{:>3}\n".format(n)
+        f.write(mode_str)
+
+    f.write("C\n")
+    f.write("C   NEV NITMAX         AL0     EPSPAM IGREEN MPINIT\n")
+    f.write("      {}   4500  {:10.3e}  1.000E-04      0      0\n".format(nev, al0))
+    f.write("C\n")
+
+    f.close()
+
+
+def _pool_fun(k, path, exec, scalar, surfs, al0, sleep_time, stop_time):
```

Comment:
what is scalar?

Reply:
I have removed this, I was testing a method to optimize the dW values on each surface instead of the scalar growth rate.